File: src/models/Activity.js
export class Activity{
    constructor(id, name, date, time, location, description, estimatedCost, status, tripId){
        this.id = id;
        this.name = name;
        this.date = date;
        this.time = time;
        this.location = location;
        this.description = description;
        this.estimatedCost = estimatedCost;
        this.status = status;
        this.tripId = tripId;
    }

}

File: TravelPlanner.Backend/ActivityService/Models/Activity.cs
using Common.Enums;
using System.ComponentModel.DataAnnotations;

namespace ActivityService.Models
{
    public class Activity
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; } = string.Empty;
        [Required]
        public DateTime Date { get; set; }

        public string? Time { get; set; }
        public string? Location { get; set; }
        public string? Description { get; set; }
        public decimal? EstimatedCost { get; set; }
        public ActivityStatus Status { get; set; } = ActivityStatus.planned;
        public int TripId { get; set; }
    }
}
